extensions/wikia/UserLogin/UserLoginSpecialController.class.php

```diff
@@ -114,286 +114,86 @@ private function getAuthenticationResource() {
 	 * Shown for both Special:UserLogin and Special:UserSignup when visited logged in.
 	 */
 	public function loggedIn() {
-		// don't show "special page" text
-		$this->wg->SupressPageSubtitle = true;
-		$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_MUSTACHE );
-
-		$userName = $this->wg->user->getName();
-		$mainPage = Title::newMainPage()->getText();
-		$userPage = Title::newFromText( $userName, NS_USER )->getFullText();
-
-		$title = wfMessage( 'userlogin-logged-in-title' )
-			->params( $userName )
-			->text();
-		$message = wfMessage( 'userlogin-logged-in-message' )
-			->params( $mainPage, $userPage )
-			->parse();
-
-		$this->wg->Out->setPageTitle( $title );
-		$this->message = $message;
+		$this->redirectToMercury();
 	}
 
-	/**
-	 * Serves standalone login page on GET. If POSTed, parameters will be required.
-	 *   on GET, template will render
-	 *   on POST,
-	 *     if login is successful, it will redirect to returnto, or default login welcome screen
-	 *     if login is not successful, the template will render will error messages, highlighting the errors
-	 * @requestParam string username - on POST
-	 * @requestParam string password - on POST
-	 * @requestParam string keeploggedin [true/false] - on POST
-	 * @requestParam string returnto - url to return to upon successful login
-	 * @requestParam string loginToken
-	 * @requestParam string action
-	 * @responseParam string result [ok/error/unconfirm/resetpass]
-	 * @responseParam string msg - result message
-	 * @responseParam string errParam - error param
-	 * @responseParam string editToken - token for changing password
-	 */
 	public function loginForm() {
```

Comment:
This method is also handling POSTs for change password and lock account, therefore it redirects to Mercury only if does not match one of these.